File: src/main/java/com/alexandre/splittrip/model/Trip.java
package com.alexandre.splittrip.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
public class Trip {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @Column(unique = true, nullable = false)
    private String roomCode;

    private LocalDateTime createdAt = LocalDateTime.now();

    @OneToMany(mappedBy = "trip", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Member> members;

    public Trip() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getRoomCode() { return roomCode; }
    public void setRoomCode(String roomCode) { this.roomCode = roomCode; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
    public List<Member> getMembers() { return members; }
    public void setMembers(List<Member> members) { this.members = members; }
}

File: mobile/lib/models/trip.dart
import 'member.dart';

class Trip {
  final int id;
  final String name;
  final String roomCode;
  final List<Member> members;

  Trip({
    required this.id,
    required this.name,
    required this.roomCode,
    required this.members,
  });

  factory Trip.fromJson(Map<String, dynamic> json) {
    return Trip(
      id: json['id'],
      name: json['name'] ?? '',
      roomCode: json['roomCode'] ?? '',

      members: json['members'] != null
          ? (json['members'] as List).map((i) => Member.fromJson(i)).toList()
          : [],
    );
  }
}
